spec/lib/msf/core/exploit/remote/remote/http/http_cookie_spec.rb

```diff
@@ -397,6 +397,108 @@ def random_string(min_len = 1, max_len = 12)
     end
   end
 
+  describe 'origin' do
+    describe 'String' do
+      it 'is assigned to origin successfully' do
+        n = random_string
+
+        cookie.origin = n
+
+        expect(cookie.origin).to eql(URI(n))
+        expect(cookie.origin.class).to eql(URI::Generic)
+      end
+
+      it 'that is empty is passed to name and throws an ArgumentError' do
+        n = ''
+
+        expect do
+          cookie.name = n
+        end.to raise_error(ArgumentError)
+      end
+
+      it 'representing a URI will be passed to origin successfully, and populate the domain & path fields correctly' do
+        domain = random_string(5)
+        path = "/#{random_string(5)}/"
+        protocol = "http#{rand(2) == 1 ? 's' : ''}"
+        my_str = "#{protocol}://#{domain}#{path}"
+
+        cookie.origin = my_str
+
+        expect(cookie.origin).to eql(URI(my_str))
+        expect(cookie.domain).to eql(domain)
+        expect(cookie.path).to eql(path)
+        expect(cookie.origin.class).to eql(protocol.end_with?('s') ? URI::HTTPS : URI::HTTP)
```

Comment:
Let's add two separate tests for this, instead of introducing non-deterministic tests :+1: